src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -376,33 +373,19 @@ private ChatCompletionsOptions ToAzureAIOptions(IList<ChatMessage> chatContents,
     /// <summary>Converts an Extensions function to an AzureAI chat tool.</summary>
     private static ChatCompletionsToolDefinition ToAzureAIChatTool(AIFunction aiFunction)
     {
-        BinaryData resultParameters = AzureAIChatToolJson.ZeroFunctionParametersSchema;
-
-        var parameters = aiFunction.Metadata.Parameters;
-        if (parameters is { Count: > 0 })
+        BinaryData functionParameters = AzureAIChatToolJson.ZeroFunctionParametersSchema;
+        if (aiFunction.Metadata.Schema is { } schema)
```

Comment:
It's no longer nullable, right? Why is this check needed?

Reply:
I did a last minute change to non-nullable but forgot to update the leaf clients since it didn't warn. Removed the dead code.